test/helpers/TaskQueue.spec.js

```diff
@@ -0,0 +1,145 @@
+const assert = require('assert');
+
+// Import class to test
+const TaskQueue = require('../../helpers/TaskQueue');
+
+describe('helpers > TaskQueue', async function () {
+  it('Resolves add method once task is complete', async function () {
+    const queue = new TaskQueue(1);
+
+    let foo;
+    const task = async () => {
+      // Timeout for 10 milliseconds
+      await new Promise((r) => { setTimeout(r, 10); });
+      foo = 'bar';
+      return 'baz';
+    };
+
+    const result = await queue.add({ task });
+    assert.equal(foo, 'bar', 'add method resolved before task completed');
+    assert.equal(result, 'baz', 'add method did not return result of task');
+  });
+
+  it('Runs all tasks added to a queue', async function () {
+    const queue = new TaskQueue(5);
+    const results = new Array(10).fill(false);
+    const tasks = [];
+
+    // For each array element, create a task that sets it to true and add the
+    //  task to the queue
+    for (let i = 0; i < 10; i++) {
+      const task = () => {
+        results[i] = true;
+      };
+      tasks.push(queue.add({ task }));
+    }
+
+    // Wait for all tasks to complete, and check if all array elements are true
+    await Promise.all(tasks);
+    assert(results.every((e) => { return e; }), 'not all tasks completed');
+  });
+
+  it('Rejects all pending tasks', async function () {
+    const queue = new TaskQueue(5);
+    let count = 0;
+    const tasks = [];
+
+    const task = () => {
+      count += 1;
+      if (count === 5) {
+        queue.rejectAll(new Error('Queue rejected'));
+      }
+    };
+
+    for (let i = 0; i < 10; i++) {
+      tasks.push(queue.add({ task }));
+    }
+
+    try {
+      await Promise.all(tasks);
+      throw new Error('Queue did not reject');
+    } catch (err) {
```

Comment:
It seems we're checking to make sure that at least one task rejects, not that all the tasks reject.

Reply:
That's true here, although I do check that all promises resolve and exactly 5 tasks complete successfully, so I think that ensures that all others also reject. I can try to check more directly if that would be preferable